internal/app/wwctl/container/exec/main.go

```diff
@@ -154,3 +175,56 @@ func SetBinds(myBinds []string) {
 func SetNode(myNode string) {
 	nodeName = myNode
 }
+
+// file name and last modification time so we can remove the file if it wasn't modified
+type copyFile struct {
+	fileName string
+	src      string
+	modTime  time.Time
+}
+
+func (this *copyFile) copyToContainer(containerName string) error {
+	containerDest := path.Join(container.RootFsDir(containerName), this.fileName)
+	if _, err := os.Stat(path.Dir(containerDest)); err != nil {
+		return fmt.Errorf("destination directory doesn't exist: %s", err)
+	} else if _, err := os.Stat(containerDest); err == nil {
+		return fmt.Errorf("file already exists in container: %s", this.fileName)
+	} else if _, err := os.Stat(this.src); err != nil {
+		return fmt.Errorf("source doesn't exist: %s", err)
+	} else if err := util.CopyFile(this.src, containerDest); err != nil {
+		return fmt.Errorf("couldn't copy files into container: %w", err)
+	} else {
+		// we can ignore error as the file was copied
+		stat, _ := os.Stat(containerDest)
+		this.modTime = stat.ModTime()
+		return nil
+	}
+}
+
+func (this *copyFile) removeFromContainer(containerName string) error {
+	containerDest := path.Join(container.RootFsDir(containerName), this.fileName)
+	if this.modTime.IsZero() {
+		return fmt.Errorf("not previously copied: %s", this.fileName)
+	} else if destStat, err := os.Stat(containerDest); err != nil {
+		return fmt.Errorf("no longer present: %s", err)
+	} else if destStat.ModTime() == this.modTime {
+		return os.Remove(containerDest)
+	} else {
+		return fmt.Errorf("modified since copy: %s", this.fileName)
```

Comment:
This isn't an error. So we should inform the user about that the file was modified, but keeping the modified file in the node image is the intended purpose.

Reply:
> This isn't an error.

I think I agree with you; this is just my Python brain over-identifying golang errors with exceptions, which they aren't really.

I'll move the more expected parts of these out of Remove.